src/github/handlers/issue-comment-created.ts

```diff
@@ -80,70 +236,12 @@ async function commandRouter(context: GitHubContext<"issue_comment.created">) {
     }
   }
 
-  const response = await context.openAi.chat.completions.create({
-    model: "gpt-4o-mini",
-    messages: [
-      {
-        role: "system",
-        content: [
-          {
-            text: `
-You are a GitHub bot named **UbiquityOS**. Your role is to interpret and execute commands based on user comments provided in structured JSON format.
-
-### JSON Structure:
-The input will include the following fields:
-- repositoryOwner: The username of the repository owner.
-- repositoryName: The name of the repository where the comment was made.
-- issueNumber: The issue or pull request number where the comment appears.
-- author: The username of the user who posted the comment.
-- comment: The comment text directed at UbiquityOS.
-
-### Example JSON:
-{
-  "repositoryOwner": "repoOwnerUsername",
-  "repositoryName": "example-repo",
-  "issueNumber": 42,
-  "author": "user1",
-  "comment": "@UbiquityOS please allow @user2 to change priority and time labels."
-}
+  // Get similar examples for the current input
+  const similarExamples = await findSimilarExamples(context.voyageAiClient, context.payload.comment.body.trim());
```

Comment:
What's the rationale here?

Reply:
Since embeddings are free for us, rather than storing them, we extract all the examples along with their commands and generate their embeddings. We then rank them using cosine similarity and select the top-k (default 3) for the prompt. If they don't match the intended commands, we can update the manifest by adding more examples.